Add overview of known espeasy nodes to json

weisskopfjens opened the pull request:
Adds a overview of known espeasy nodes to the json response. I think this can be useful for own webinterfaces etc.

weisskopfjens committed:
Adds a overview of known espeasy nodes to the json response. I think this can be useful for own webinterfaces etc.

Grovkillen commented:
Nice addition, we'll review the code and implementation.

TD-er commented on src/WebServer.ino:
```diff
@@ -3798,9 +3800,70 @@ void handle_json()
+          switch (Nodes[x].nodeType)
+          {
+            case NODE_TYPE_ID_ESP_EASY_STD:
+              stream_next_json_object_value(F("type"), F("ESP Easy"));
```
It is better to have the second value stored in a temporary `String`, since those `F("type")` macro strings will appear multiple times in the bin file, if done like this.
Also I am not sure "type" is the best description for this. I guess "platform" is a better choice.

TD-er commented:
There is still a bug when the UDP port is not set.
Then it will not output valid JSON.
I will make a patch for it and append to this PR.

TD-er committed:
[JSON] Fix issue when UDP port not set

With no nodes to display, the generated JSON was not valid.
Also removed some duplicate F() macro strings.

TD-er committed:
[JSON] Only show node information when generic info is requested

This extra node information is not needed when values of a specific task is requested, so only output when other generic information is shown.
